### PR TITLE
Adding dataset.delete(obj), which calls the specific type's delete

### PR DESCRIPTION
**Pull request by maxhutch**
# Citrine Python PR

## Description 
This is a convenience method that makes it much easier to delete
data objects in loops.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

File: tests/resources/test_dataset.py [hunk 1]
```diff
@@ -346,3 +346,24 @@ def test_register_all_object_update(dataset):
...
+    for collection, obj in expected.items():
+        dataset.delete(obj)
+        assert dataset.session.calls[-1].path.split("/")[-3] == basename(collection._path_template)
+
+
+def test_delete_missing_uid(dataset):
+    """Check that delete raises an error when there are no uids"""
+    obj = MaterialTemplate("foo")
+    with pytest.raises(ValueError):
+        dataset.delete(obj)
```
Comment (kroenlein) [inline tests/resources/test_dataset.py:369]:
```suggestion
        dataset.delete(obj)

```
Missing trailing newline

File: src/citrine/resources/dataset.py [hunk 2]
```diff
@@ -271,6 +271,14 @@ def register_all(self, data_concepts_resources: List[ResourceType],
...
         return resources
 
+    def delete(self, data_concepts_resource: ResourceType, dry_run=False) -> ResourceType:
+        """Delete a data concepts resource to the appropriate collection."""
+        uid = next(iter(data_concepts_resource.uids.items()), None)
+        if uid is None:
+            raise ValueError("Only objects that contain identifiers can be deleted.")
+        return self._collection_for(data_concepts_resource) \
+            .delete(uid[1], scope=uid[0], dry_run=dry_run)
+
```
Comment (kroenlein) [inline src/citrine/resources/dataset.py:281]: Makes sense and is generally consistent with how methods on Dataset work.

File: tests/resources/test_dataset.py (hunk 1 above)
Comment (maxhutch) [reply tests/resources/test_dataset.py:369]: Wouldn't flake have picked this up?

Comment (maxhutch) [reply tests/resources/test_dataset.py:369]: nvm; test source.  :shrug:

**Pull request merged**